eta/core/c3d.py

```diff
@@ -0,0 +1,340 @@
+
+        # Convolution Layer
+        conv2 = self.conv3d('conv2', pool1, _weights['wc2'], _biases['bc2'])
+        conv2 = tf.nn.relu(conv2, 'relu2')
+        pool2 = self.max_pool('pool2', conv2, k=2)
+
+        # Convolution Layer
+        conv3 = self.conv3d('conv3a', pool2, _weights['wc3a'], _biases['bc3a'])
+        conv3 = tf.nn.relu(conv3, 'relu3a')
+        conv3 = self.conv3d('conv3b', conv3, _weights['wc3b'], _biases['bc3b'])
+        conv3 = tf.nn.relu(conv3, 'relu3b')
+        pool3 = self.max_pool('pool3', conv3, k=2)
+
+        # Convolution Layer
+        conv4 = self.conv3d('conv4a', pool3, _weights['wc4a'], _biases['bc4a'])
+        conv4 = tf.nn.relu(conv4, 'relu4a')
+        conv4 = self.conv3d('conv4b', conv4, _weights['wc4b'], _biases['bc4b'])
+        conv4 = tf.nn.relu(conv4, 'relu4b')
+        pool4 = self.max_pool('pool4', conv4, k=2)
+
+        # Convolution Layer
+        conv5 = self.conv3d('conv5a', pool4, _weights['wc5a'], _biases['bc5a'])
+        conv5 = tf.nn.relu(conv5, 'relu5a')
+        conv5 = self.conv3d('conv5b', conv5, _weights['wc5b'], _biases['bc5b'])
+        conv5 = tf.nn.relu(conv5, 'relu5b')
+        pool5 = self.max_pool('pool5', conv5, k=2)
+
+        # Fully connected layer
+        pool5 = tf.transpose(pool5, perm=[0,1,4,2,3])
+        dense1 = tf.reshape(pool5, [1, _weights['wd1'].get_shape().as_list()[0]]) # Reshape conv3 output to fit dense layer input
+        dense1 = tf.matmul(dense1, _weights['wd1']) + _biases['bd1']
+
+        self.fc6 = tf.nn.relu(dense1, name='fc1') # Relu activation
+        dense1 = tf.nn.dropout(self.fc6, _dropout)
+
+        dense2 = tf.nn.relu(tf.matmul(dense1, _weights['wd2']) + _biases['bd2'], name='fc2') # Relu activation
+        dense2 = tf.nn.dropout(dense2, _dropout)
+
+        # Output: class prediction
+        self.out = tf.matmul(dense2, _weights['out']) + _biases['out']
+
+    def _load_model(self):
+        saver = tf.train.Saver()
+        init = tf.global_variables_initializer()
+        self.sess.run(init)
+        saver.restore(self.sess, self.config.model_path)
+
+
+class C3DFeaturizerConfig(C3DConfig):
+    ''' C3D Featurization configuration settings that works on images'''
+    pass
+
+
+class C3DFeaturizer(Featurizer):
+    def __init__(self, config=None):
+        super(C3DFeaturizer,self).__init__()
+        self.config = config or C3DFeaturizerConfig.default()
+        self.validate(self.config)
+        self.sample_method = self.config.sample_method
+        self.c3d  = None
+
+    def get_first_k_frames(self, inpath, k, out_size):
+        '''Sample first k frames in a video.
+
+        Args:
+            inpath: path to the input video
+            k: number of frames to extract for the input video
+            out_size: size of the output frames
+        '''
+        data = []
+        num_frames = etav.get_frame_count(inpath)
+        assert k < num_frames
+        with etav.VideoProcessor(inpath, "1-%d" % k, out_size=out_size) as p:
```

Comment:
`VideoProcessor` is for reading a video and then writing a new video in response. You should use `FFmpegVideoReader` here because you're just reading a video